prime-router/src/main/kotlin/history/azure/DatabaseDeliveryAccess.kt

```diff
@@ -141,4 +155,112 @@ class DatabaseDeliveryAccess(
+     * @todo further work on https://github.com/CDCgov/prime-reportstream/issues/9272
+     * @param reportIds list of reports to check facilities for
+     * @return List of matching delivery facilities
+     */
+    fun fetchBulkDeliveryFacilities(
+        receiver: String,
+        receivingOrgSvc: String?,
+        sortDir: SortDir,
+        sortColumns: List<BulkFacilitySortColumn>,
+        since: OffsetDateTime?,
+        until: OffsetDateTime?,
+        pageSize: Int,
+        pageNumber: Int
+    ): List<DeliveryFacility> {
+        val reportIds = db.transactReturning { txn ->
+            var filter = this.organizationFilter(receiver, receivingOrgSvc)
+            if (since != null) {
+                filter = filter.and(ACTION.CREATED_AT.ge(since))
+            }
+            if (until != null) {
+                filter = filter.and(ACTION.CREATED_AT.lt(until))
+            }
+
+            val query = DSL.using(txn)
+                // Note the report file and action tables have columns with the same name, so we must specify what we need.
+                .select(REPORT_FILE.REPORT_ID)
+                .from(
+                    ACTION.join(REPORT_FILE).on(
+                        REPORT_FILE.ACTION_ID.eq(ACTION.ACTION_ID)
+                    )
+                )
+                .where(filter)
+                .groupBy(REPORT_FILE.REPORT_ID)
+
+            query.fetchInto(ReportId::class.java)
+        }
+
+        return db.transactReturning { txn ->
+            var filter = COVID_RESULT_METADATA.REPORT_ID.isNotNull()
+            if (since != null) {
+                filter = filter.and(COVID_RESULT_METADATA.CREATED_AT.ge(since.toLocalDateTime()))
+            }
+            if (until != null) {
+                filter = filter.and(COVID_RESULT_METADATA.CREATED_AT.lt(until.toLocalDateTime()))
+            }
+
+            val sortedColumns = sortColumns.map {
+                val column = when (it) {
+                    /* Decides sort column by enum */
+                    BulkFacilitySortColumn.DATE -> ITEM_LINEAGE.CREATED_AT
+                    BulkFacilitySortColumn.PROVIDER -> COVID_RESULT_METADATA.ORDERING_PROVIDER_NAME
+                    BulkFacilitySortColumn.FACILITY -> COVID_RESULT_METADATA.TESTING_LAB_NAME
+                    BulkFacilitySortColumn.SUBMITTER -> COVID_RESULT_METADATA.SENDER_ID
+                    BulkFacilitySortColumn.REPORT_ID -> COVID_RESULT_METADATA.REPORT_ID
+                }
+
+                when (sortDir) {
+                    /* Applies sort order by enum */
+                    SortDir.ASC -> column.asc()
+                    SortDir.DESC -> column.desc()
+                }
+            }
+            val positive: Field<String> = DSL.field(
+                "sum(CASE WHEN test_result = 'DETECTED' THEN 1 ELSE 0 END)",
+                String::class.java
+            ).`as`("positive")
+            val total: Field<String> = DSL.field(
```

Comment:
I don't think this is a field in the mockup?

Reply:
This is another bad copy field we can get rid of for this call